Source/monster.cpp

```diff
@@ -4515,40 +4516,24 @@ void SyncMonsterAnim(Monster &monster)
 
 void M_FallenFear(Point position)
 {
-	for (int i = 0; i < ActiveMonsterCount; i++) {
-		auto &monster = Monsters[ActiveMonsters[i]];
-
-		if (monster._mAi != AI_FALLEN)
+	const Rectangle fearArea = Rectangle { position, 4 };
+	for (const Point tile : PointsInRectangleRange { fearArea }) {
+		if (!InDungeonBounds(tile))
 			continue;
-		if (position.WalkingDistance(monster.position.tile) >= 5)
+		int m = dMonster[tile.x][tile.y];
+		if (m == 0)
 			continue;
-		if (monster._mhitpoints >> 6 <= 0)
+		Monster &monster = Monsters[m - 1];
+		if (monster._mAi != AI_FALLEN || monster._mhitpoints >> 6 <= 0)
 			continue;
 
-		int rundist;
-		switch (monster.MType->mtype) {
-		case MT_RFALLSP:
-		case MT_RFALLSD:
-			rundist = 7;
-			break;
-		case MT_DFALLSP:
-		case MT_DFALLSD:
-			rundist = 5;
-			break;
-		case MT_YFALLSP:
-		case MT_YFALLSD:
-			rundist = 3;
-			break;
-		case MT_BFALLSP:
-		case MT_BFALLSD:
-			rundist = 2;
-			break;
-		default:
-			continue;
-		}
+		int intelligenceFactor = monster._mint;
+		if (monster.MType->mtype == MT_DFALLSP)
+			intelligenceFactor--;
+		int runDistance = std::max((7 - 2 * intelligenceFactor), 2);

```

Comment:
On closer inspection this might cause a change in behavior since Pukerat the Unclean, Bongo, Gutshank the Quick, Shadowcrow and there minions have altered intelligence compared to the base.

Reply:
Oh gee, I really liked this :)
Btw I guess this is making them harder a bit ;)
I will try to fry some solution for this.
Btw it is only Pukerat the Unclean and Bongo.
Since Gutshank the Quick has AI_BAT, and Shadowcrow AI_SNEAK. (And Bladeskin the Slasher is ok with 0-0 and Snotspil 3-3)

Reply:
Pukerat the Unclean easy - no mobs

Reply:
lol, got better idea
Fallen level 1
Carver level 3
Devil Kin level 5
Dark One level 7
:)
how nice is that

Reply:
unless it have to be otherwise :)
Fallen level 1 -> 7
Carver level 3 -> 5
Devil Kin level 5 -> 3
Dark One level 7 -> 2

Reply:
`max (8 - base_monster_level), 2)`

Reply:
That might work